tools/offline_ota_image_builder/builder.py

```diff
@@ -0,0 +1,221 @@
+# Copyright 2022 TIER IV, INC. All rights reserved.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+
+import logging
+import os
+import pprint
+import shutil
+import tempfile
+import time
+from contextlib import contextmanager
+from pathlib import Path
+from typing import Mapping, Optional, Sequence
+
+from otaclient.app.common import subprocess_call
+from otaclient.app.ota_metadata import parse_regulars_from_txt
+
+from .configs import cfg
+from .manifest import ImageMetadata, Manifest
+from .utils import StrPath, InputImageProcessError, ExportError
+
+logger = logging.getLogger(__name__)
+
+PROC_MOUNTS = "/proc/mounts"
+
+
+def _check_if_mounted(dev: StrPath):
+    for line in Path(PROC_MOUNTS).read_text().splitlines():
+        if line.find(str(dev)) != -1:
+            return True
+    return False
+
+
+@contextmanager
+def _unarchive_image(image_fpath: StrPath, *, workdir: StrPath):
+    _start_time = time.time()
+    with tempfile.TemporaryDirectory(dir=workdir) as unarchive_dir:
+        cmd = f"tar xf {image_fpath} -C {unarchive_dir}"
+        try:
+            subprocess_call(cmd)
```

Comment:
why do you use subprocess, you can use tar python library.

Reply:
@obi-t4 python tar stdlib is very slow as it is written in pure python.